trino-aws-proxy-spi/src/main/java/io/trino/aws/proxy/spi/credentials/Credentials.java

```diff
@@ -19,13 +19,14 @@
 
 import static java.util.Objects.requireNonNull;
 
-public record Credentials(Credential emulated, Optional<Credential> remote, Optional<RemoteSessionRole> remoteSessionRole)
+public record Credentials(Credential emulated, Optional<Credential> remote, Optional<RemoteSessionRole> remoteSessionRole, Optional<Identity> identity)
```

Comment:
Alternatively, we could make this a parameterized type. I'm not sure which is better, though. wdyt?